spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt

```diff
@@ -58,4 +52,19 @@ actual class SpekRuntime: AbstractRuntime() {
             .map { it.objectInstance ?: it.primaryConstructor!!.call() }
             .firstOrNull() ?: defaultInstanceFactory
     }
+
+    private fun createReflections(testDirs: List<String>): Reflections {
+        val urls = if (testDirs.isEmpty()) {
+            ClasspathHelper.forClassLoader()
+        } else {
+            testDirs.map { java.nio.file.Paths.get(it) }
+                .map { it.toUri().toURL() }
+        }
+
+        return Reflections(
```

Comment:
BTW, why Spek does not use the native Kotlin relfection for this? Or it is too limited for Spek needs?

Reply:
Not limited per se, you just need to do a lot of manual tedious hacking. `Reflections` does it clear and neat way. Basically, less things to think about.